sources/ec2/instance.go

```diff
@@ -61,6 +71,23 @@ func instanceOutputMapper(_ context.Context, _ *ec2.Client, scope string, _ *ec2
 				},
 			}
 
+			if instance.State != nil {
+				switch aws.ToInt32(instance.State.Code) {
+				case codeRunning:
+					item.Health = sdp.Health_HEALTH_OK.Enum()
+				case codePending:
+					item.Health = sdp.Health_HEALTH_PENDING.Enum()
+				case codeShuttingDown:
+					item.Health = sdp.Health_HEALTH_PENDING.Enum()
+				case codeTerminated:
+					item.Health = sdp.Health_HEALTH_ERROR.Enum()
```

Comment:
Using error state when it is terminated doesn't sound great.
Should there be a state for known terminated resources?